src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/gencode/GencodeFuncotationFactory.java

```diff
@@ -177,25 +192,36 @@ public GencodeFuncotationFactory(final Path gencodeTranscriptFastaFile,
             this.userRequestedTranscripts.add( getTranscriptIdWithoutVersionNumber(transcript) );
         }
 
+        // Set our comparator for outputting our funcotations in the right order with the correct "best" transcript:
+        if ( transcriptSelectionMode == FuncotatorArgumentDefinitions.TranscriptSelectionMode.BEST_EFFECT ) {
```

Comment:
I would also tend to bake the comparator into the TranscriptSelectionMode enum itself, so instead of this if statement you just call `transcriptSelectMode.getComparator(userRequestedTranscripts)`

Reply:
Interesting.  Sounds good.